diff -ruN -x .git old/src/core/nxlog.c new/src/core/nxlog.c
--- old/src/core/nxlog.c	2023-11-17 09:05:23.000000000 +0000
+++ new/src/core/nxlog.c	2025-09-29 06:31:10.334158743 +0000
@@ -58,7 +58,14 @@
     nx_ctx_shutdown_modules(nxlog->ctx, NX_MODULE_TYPE_PROCESSOR);
     nx_ctx_shutdown_modules(nxlog->ctx, NX_MODULE_TYPE_OUTPUT);
     nx_ctx_shutdown_modules(nxlog->ctx, NX_MODULE_TYPE_EXTENSION);
-    nx_config_cache_write();
+    if ( nxlog->verify_conf == FALSE )
+    {
+        nx_config_cache_write();
+    }
+    else
+    {
+        log_debug("nxlog_shutdown() skip config cache write on configuration verify");
+    }
     nx_config_cache_free();
     log_debug("nxlog_shutdown() leave");
 }
diff -ruN -x .git old/src/modules/input/systemd/im_systemd.c new/src/modules/input/systemd/im_systemd.c
--- old/src/modules/input/systemd/im_systemd.c	2023-11-17 09:05:23.000000000 +0000
+++ new/src/modules/input/systemd/im_systemd.c	2025-09-29 06:31:10.334406272 +0000
@@ -134,6 +134,7 @@
     im_systemd_config_t *conf;
     int err;
     char *cur;
+    int skipped;
 
     ASSERT(module != NULL);
     ASSERT(module->config != NULL);
@@ -169,6 +170,24 @@
 	CHECKERR_NEG_MSG(sd_journal_next(conf->journal),
 	                 "Skip last message");
     }
+    else
+    {
+        log_warn("Couldn't get cursor from config cache. Read from end");
+        CHECKERR_NEG_MSG(sd_journal_seek_tail(conf->journal),
+            "Couldn't seek at tail record");
+
+        skipped = 0;
+        while ( sd_journal_next(conf->journal) > 0 )
+        {
+            skipped++;
+        }
+        if ( skipped != 0 ) {
+            log_debug("Last messages (%d) skipped to seek at tail record", skipped);
+        }
+
+        CHECKERR_NEG_MSG(sd_journal_previous(conf->journal),
+            "Couldn't seek at last record");
+    }
 }
 
 
